src/Microsoft.DotNet.Build.CloudTestTasks/FilterForUpload.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    public sealed class FilterForUpload : Task
+    {
+        //This task is used to select the appropriate test zips from the different output directories produced by the build,
+        //which are created based on OSGroup and TestTFM.
+
+        [Required]
+        public string TestOutputDirListsLocation { get; set; }
+
+        //This is the TestTFM folder that we want to select test zips from. This needs to be on the path to the test zip.
+        [Required]
+        public string FilterToTestTFM { get; set; }
+
+        //This is the OSGroup folder that we want to select test zips from. This does not require an exact match as some tests 
+        //may or may not have an exact OSGroup folder match.
+        [Required]
+        public string FilterToOSGroup { get; set; }
+        //This is the path under a given Platform/OSGroup/Configuration folder where the Test Archives are placed
+        [Required]
+        public string TestArchivesRoot { get; set; }
+
+        [Output]
+        public ITaskItem[] FilteredUploadItems
+        {
+            get { return this._copyFilteredUploadItems.ToArray(); }
```

Comment:
This field should be named the same as the property (or, better yet, just us an auto-prop and assign it at the end of Execute)